test/integration/crud/misc_cursors.test.js

```diff
@@ -3841,107 +3063,93 @@ describe('Cursor', function () {
     });
   });
 
-  it('should not consume first document on hasNext when streaming', function (done) {
+  it('should not consume first document on hasNext when streaming', async function () {
```

Comment:
suggest removing - very niche method, we have adequate coverage for both the stream() method and the hasNext() individually.